docs/tutorials/send-payjoin-with-pdk.md

```diff
@@ -0,0 +1,311 @@
+on crafting and sending your very own Payjoin transactions. Moreover, We'll
+guide you through setting up payjoin-cli, and use it a the receiver side in the
+payjoin transaction.
+
+# Bitcoin Node & Wallets
+  In order to perform a payjoin transaction, the following code will create two
+  bitcoin wallets, "sender" and "receiver" connected to bitcoin node running in
+  regtest mode.
+
+  * You need to have docker installed!
+
+  ```shell
+  cargo install pjregtest
+  ```
+
+  ```shell
+  pjregtest
+  ```
+
+# Project Setup
+
+  Lets setup a new rust project and name it `payjoin-tutorial`
+  ```shell
+	cargo new payjoin-tutorial
+  ```
+  
+  Add required dependecies to `cargo.toml` so it looks like the
+  following:
+
+  ```toml
+  [dependencies]
+  bitcoincore-rpc = "0.17.0" 
+  payjoin = { version = "0.13.0", features = ["send", "base64"] }
+  reqwest = {version = "0.11.24", features = ["rustls-tls"]}
+  tokio = {version = "1.36.0", features = ["full"]}
+  ```
+
+# Sender
+
+  In order to send a payjoin transaction, the sender need to complete two steps.
+
+  First, obtains the reciever BIP21 URI(usually via qr-code), construct a PSBT
+  based on paramaters, like address, payjoin endpoint(pj) and amount, as
+  specified in the URI.
+
+  Second, the sender construct an HTTPs request, with the PSBT attached to the
+  body. After sending the request, the sender waits for the response and when it
+  arrives, its verified, signed and broadcasted to the bitcoin blockchain.
+
+
+  Lets start implementing the sender functionality inside
+  `payjoin-tutorial/src/main.rs`
+
+  * Add imports:
+
+  ```rust
+    use bitcoincore_rpc::RpcApi;
+    use payjoin::bitcoin::address::NetworkChecked;
+    use std::collections::HashMap;
+    use std::str::FromStr;
+
+    use bitcoincore_rpc::bitcoin::psbt::Psbt;
+    use bitcoincore_rpc::bitcoin::Txid;
+    use payjoin::base64;
+    use payjoin::send::RequestBuilder;
+   ```
+
+  * Add an `https_agent` function. 
+
+    As you would guess, the `https_agent` function is used in order to make
+    https requests, the sender will use this function to send the payjoin
+    request to the reciever.
+
+    Here, we are using `danger_accept_invalid_certs` as we running this locally,
+    in production this should be initiated with valid certificates only.
+
+```rust
+ fn https_agent() -> reqwest::Client {
+     let https = reqwest::Client::builder()
+         .danger_accept_invalid_certs(true)
+         .build()
+         .expect("failed to build https client");
+     https
+ }
+```
+
+  * Add `Sender` struct
+    
+    We will add an implementation to the struct with a single function,
+    `try_payjoin`. The function will take a `payjoin::Uri` as an input and
+    attempt to perform a payjoin transaction.
+    
+```rust
+struct Sender;
+
+impl Sender {
+    pub async fn try_payjoin(pj_uri: payjoin::Uri<'static, NetworkChecked>) -> Txid {
+    }
+}
+```
```

Comment:
if the sender doesn't hold state, you're better off either making a `mod sender` instead of a struct or just leaving the functions in the main body of the file.